src/components/InputTypes/Code.tsx

```diff
@@ -1,11 +1,37 @@
 /** @jsx jsx */
 import { jsx } from "@emotion/react";
-import React, { ChangeEvent } from "react";
+import React, { useEffect } from "react";
+
+import { basicSetup } from "@codemirror/basic-setup";
+import { python } from "@codemirror/lang-python";
+import { EditorState } from "@codemirror/state";
+import { oneDark } from "@codemirror/theme-one-dark";
+import { EditorView, ViewUpdate } from "@codemirror/view";
 
 interface CodeProps {
-    handler: (event: ChangeEvent<HTMLInputElement>) => void
+    handler: (newContent: string) => void,
+    questionId: string,
 }
 
 export default function Code(props: CodeProps): JSX.Element {
-    return <input type="text" className="text" onChange={props.handler}/>;
+    const onUpdate = () => EditorView.updateListener.of((v: ViewUpdate) => {
+        props.handler(v.state.doc.toString());
+    });
+
+    useEffect(() => {
+        const el = document.getElementById(`${props.questionId}-code`);
+        const state = EditorState.create({
+            extensions: [basicSetup, python(), onUpdate(), oneDark]
```

Comment:
I don't like this theme at all with our background. The element blends in, and is hard to see. That being said, it seems the library doesn't provide much in terms of themes, and isn't backward compatible with older themes (I converted a few, but they don't look good), so I can live with this for now, provided we add a highly contrasting border. 

I tried `border: 3px solid lightgray; border-radius: 5px;`, and while it isn't  the prettiest thing in the world, I prefer it to what we currently have.